### PR TITLE
feat: Add stories for the useSearch hook

### PR DESCRIPTION
**Pull request by anandgorantala**
Also includes some style changes to the ProductCard to display the results better in the useSearch stories and some updates to the Theme story

### CODE REVIEW

File: src/components/theme/theme.types.ts
Comment (feixie1980) [inline src/components/theme/theme.types.ts:1]: I am actually not entirely clear about the role of <Theme>.  Can you update its use and role with comments?  This is not directly related to this review, so you can do it in another PR.

**Review by feixie1980 — APPROVED** [review]

**Review by joerideg — APPROVED** [review]
